apis/Google.LongRunning/Google.LongRunning/Operation.cs

```diff
@@ -202,14 +204,15 @@ private void MaybeFireCallback(Action<TMetadata> metadataCallback)
         {
```

Comment:
`Func<TMetadata, Task>` for this async case? Which would be awaited on when invoked.
If a user is writing async-based code, it may be frustrating that this callback is not Task-based.